spec.md

```diff
@@ -3333,6 +3350,8 @@ See https://github.com/bazelbuild/starlark/issues/47.
 
 ### float
 
+`float(x: str|bool|int|float, /) -> float`
```

Comment:
(No action needed for this PR) There may be a bunch of other args to other builtins that are Noneable, in the sense that we take `None` as an alternative to omitting the parameter entirely.

Reply:
There are bunch of None able args. Since we only have primitive types and unions at this point, I wasn't able to type those. But certainly more is coming.